target/i386/hax-mem.c

```diff
@@ -310,8 +317,17 @@ static void hax_ram_block_added(RAMBlockNotifier *n, void *host, size_t size)
     }
 }
 
+#ifdef XBOX
+static void hax_ram_block_removed(RAMBlockNotifier *n, void *host, size_t size)
+{
```

Comment:
No-op is probably wrong, but I'm not sure what to put here yet.

Reply:
- Why doesn't QEMU upstream run into this issue?
- Which handler is used in upstream QEMU?
- Did you report this upstream?
- Why not just add the NULL-check instead?

I think it's a weird Xbox specific hack which could also affect other platforms. QOM typically ignores NULL callbacks too, if I remember correctly.

Reply:
> Why doesn't QEMU upstream run into this issue?

I guess they hadn't had any RAM blocks removed in testing. We run into this when we swap out VGA RAM during NV2A basic init.

> Which handler is used in upstream QEMU?

What do you mean?

> Did you report this upstream?

No, but it should be fixed upstream. I'll probably report it in the coming days.

> Why not just add the NULL-check instead?

I've run into this problem with other QEMU code too--mostly due to interface changes. I also think it is best practice to check for `NULL` handlers before invoking them, but in this instance I think HAXM *should* support this method, so adding the handler is appropriate.